discovery/azure/azure.go

```diff
@@ -315,30 +315,32 @@ func (d *Discovery) refresh() (tg *targetgroup.Group, err error) {
 					return
 				}
 
-				// Unfortunately Azure does not return information on whether a VM is deallocated.
-				// This information is available via another API call however the Go SDK does not
-				// yet support this. On deallocated machines, this value happens to be nil so it
-				// is a cheap and easy way to determine if a machine is allocated or not.
-				if networkInterface.Properties.Primary == nil {
-					level.Debug(d.logger).Log("msg", "Skipping deallocated virtual machine", "machine", vm.Name)
-					ch <- target{}
-					return
-				}
+				if networkInterface.Properties != nil {
```

Comment:
It'd be more usual to do a continue if it is nil, as this avoids intending all teh following.

Reply:
I thought about that but then I realized a continue might be overlooked if more code were to be added in the loop.

Reply:
If you prefer a continue I'll do it

Reply:
Yes please